src/main/java/ua/lviv/iot/parkingServer/kafka/KafkaConsumer.kt

```diff
@@ -34,16 +37,22 @@ class KafkaConsumer {
 
     @Bean
     fun kafkaConsumerTemplate(
-        kafkaReceiverOptions: ReceiverOptions<String, ByteArray>
-    ): ReactiveKafkaConsumerTemplate<String, ByteArray> {
+        kafkaReceiverOptions: ReceiverOptions<String, GeneratedMessageV3>
+    ): ReactiveKafkaConsumerTemplate<String, GeneratedMessageV3> {
         return ReactiveKafkaConsumerTemplate(kafkaConsumerOptions())
     }
 }
 
-class ProtobufDeserializer() : Deserializer<ByteArray> {
+class ProtobufDeserializer : Deserializer<GeneratedMessageV3> {
 
-    override fun deserialize(topic: String, data: ByteArray): ByteArray {
-        return data
-    }
+    private val topicToParserMap = HashMap<String, Parser<CreateParkingSpotRequest>>()
+
+    private val parser: Parser<CreateParkingSpotRequest> =
+        CreateParkingSpotRequest.parser()
 
+    override fun deserialize(topic: String, data: ByteArray): GeneratedMessageV3? {
+        topicToParserMap[topic] = parser
+        val parser = topicToParserMap[topic]
+        return parser?.parseFrom(data)
+    }
```

Comment:
```suggestion
    private val topicToParserMap = mapOf("ADDED_AVAILABLE_PARKING_SPOT" to CreateParkingSpotRequest.parser())

    override fun deserialize(topic: String, data: ByteArray): GeneratedMessageV3 {
        return topicToParserMap[topic]?.parseFrom(data) ?: throw CannotParseException()
    }
```

I meant something like this

Reply:
✅